### PR TITLE
SI-10037 ASR/LSR switched in ICodeReader

### PR DESCRIPTION
**Pull request by som-snytt**
Noticed when inlining from a class file.

JIRA: https://issues.scala-lang.org/browse/SI-10037

### CODE REVIEW

File: src/compiler/scala/tools/nsc/backend/icode/Primitives.scala [hunk 1]
```diff
@@ -210,7 +210,7 @@ trait Primitives { self: ICodes =>
       case LSL =>  "LSL"
       case ASR =>  "ASR"
       case LSR =>  "LSR"
-      case _  => throw new RuntimeException("ShitOp unknown case")
+      case _  => throw new RuntimeException("ShiftOp unknown case")
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/backend/icode/Primitives.scala:213]: 😳

**Review by lrytz — CHANGES_REQUESTED** [review]
LGTM, thanks! Could you mark the commit `[nomerge]`?

**Review by lrytz — APPROVED** [review]

**Comment by SethTisue** [comment]
```
!!  697 - run/t10037                                [output differs]
```

**Comment by som-snytt** [comment]
Thanks @SethTisue . I was a bit casual b/c home with flu; I forgot the flags. Now I see that this test isn't inlining. If anyone reads this & knows why offhand, please let me know. Maybe I'll try a `ReplTest` in case the class loaders align.

```
scala.reflect.internal.MissingRequirementError: object X in compiler mirror not found.
    at scala.reflect.internal.MissingRequirementError$.signal(MissingRequirementError.scala:17)
    at scala.reflect.internal.MissingRequirementError$.notFound(MissingRequirementError.scala:18)
    at scala.reflect.internal.Mirrors$RootsBase$$anonfun$getModuleOrClass$3.apply(Mirrors.scala:54)
    at scala.reflect.internal.Mirrors$RootsBase$$anonfun$getModuleOrClass$3.apply(Mirrors.scala:54)
    at scala.reflect.internal.Symbols$Symbol.orElse(Symbols.scala:2536)
    at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:53)
    at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:66)
    at scala.reflect.internal.Mirrors$RootsBase.getModuleByName(Mirrors.scala:134)
    at scala.tools.nsc.symtab.classfile.ClassfileParser$ConstantPool.getClassSymbol(ClassfileParser.scala:230)
    at scala.tools.nsc.symtab.classfile.ICodeReader.parseClass(ICodeReader.scala:146)
    at scala.tools.nsc.symtab.classfile.ClassfileParser$$anonfun$parse$1.apply$mcV$sp(ClassfileParser.scala:150)
    at scala.tools.nsc.symtab.classfile.ClassfileParser$$anonfun$parse$1.apply(ClassfileParser.scala:139)
    at scala.tools.nsc.symtab.classfile.ClassfileParser$$anonfun$parse$1.apply(ClassfileParser.scala:139)
    at scala.tools.nsc.symtab.classfile.ClassfileParser.pushBusy(ClassfileParser.scala:125)
    at scala.tools.nsc.symtab.classfile.ClassfileParser.parse(ClassfileParser.scala:139)
    at scala.tools.nsc.symtab.classfile.ICodeReader.readClass(ICodeReader.scala:134)
    at scala.tools.nsc.backend.icode.Repository$class.load(Repository.scala:32)
    at scala.tools.nsc.backend.icode.ICodes.load(ICodes.scala:19)
    at scala.tools.nsc.backend.opt.Inliners$Inliner.scala$tools$nsc$backend$opt$Inliners$Inliner$$analyzeInc$1(Inliners.scala:407)
```

**Comment by som-snytt** [comment]
`ReplTest` wasn't any happier. I'll experiment later. Here's the manual test, FTR:

```
$ scala211 -optimise -Yinline-warnings
Welcome to Scala 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_101).
Type in expressions for evaluation. Or try :help.

scala> shifty.X.f(Int.MinValue)
res0: Int = 1073741824

scala> shifty.X.g(Int.MinValue)
res1: Int = -1073741824

scala> :quit
$ skala -optimise -Yinline-warnings
Welcome to Scala 2.11.9-20161108-113317-74ed575a0f (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_101).
Type in expressions for evaluation. Or try :help.

scala> shifty.X.f(Int.MinValue)
res0: Int = -1073741824

scala> shifty.X.g(Int.MinValue)
res1: Int = 1073741824
```

**Comment by som-snytt** [comment]
Maybe I'll just delete the test.

**Comment by retronym** [comment]
The old optimizer [baulks](https://issues.scala-lang.org/browse/SI-8580) at the sight of the empty package. Bearing this in mind, we can [revive](https://github.com/retronym/scala/tree/review/5518) the test case.

**Comment by som-snytt** [comment]
Aha! Thanks for the hint. I got rid of packages because partest doesn't descend looking for a `Test` IIRC.

**Comment by som-snytt** [comment]
It's such a great feeling when your PR reaches the Graphviz error output.

**Pull request merged**